### PR TITLE
Enabled tests by adding opencv_extra as a lazy artifact

### PR DESCRIPTION
**Pull request by stemann**
Enabled tests to run - rather than just skipping them when opencv_extra is not in `ENV`.

### CODE REVIEW

**Comment by stemann** [comment]
Testing on x86 (on ubuntu) was disabled due to a very basic error stemming from just creating a CxxWrap `StdString` from
```julia
img_gray = OpenCV.imread(joinpath(test_dir, "shared", "pic1.png"), OpenCV.IMREAD_GRAYSCALE)
```
Cf. https://github.com/JuliaImages/OpenCV.jl/actions/runs/5725287556/job/15513486710#step:6:188

Looks like it's this: https://github.com/JuliaInterop/CxxWrap.jl/issues/310

File: test/test_feature2d.jl [hunk 1]
```diff
@@ -17,7 +17,5 @@ for kp in kps
         end
     end
 
-    @test closest_match < 10
+    @test_skip closest_match < 10 # TODO This test fails intermittently
```
Comment (ashwani-rathee) [inline test/test_feature2d.jl:20]: Coul we fix this too in this round?

**Review by ashwani-rathee — CHANGES_REQUESTED** [review]
Otherwise looks alright, do we need specifically LazyArtifacts for making artifact download lazy?

File: test/test_feature2d.jl (hunk 1 above)
Comment (stemann) [reply test/test_feature2d.jl:20]: No - it's not immediately apparent how to fix this test - except for replacing it completely with a test for SimpleBlobDetector from OpenCV, e.g. https://github.com/opencv/opencv/blob/4.x/modules/features2d/misc/java/test/SIMPLEBLOBFeatureDetectorTest.java

Also note [comment in #13](https://github.com/JuliaImages/OpenCV.jl/issues/13#issuecomment-1661637648). 

For this PR, I would like to just focus on getting the tests enabled.

Comment (ashwani-rathee) [reply test/test_feature2d.jl:20]: Okay, I could see why it's a bit problematic in current state. Let's merge and keep a note of the java test in the #13 open

**Pull request merged**